packages/ui/src/multiplying-architecture/KaotoEditorChannelApi.ts

```diff
@@ -12,4 +12,32 @@ export interface KaotoEditorChannelApi extends KogitoEditorChannelApi {
    * Returns the Kaoto VSCode settings defined.
    */
   getVSCodeKaotoSettings(): Promise<ISettingsModel>;
+
+  /**
+   * Get metadata querying a Kaoto metadata file.
+   * @param path The path of the file being edited
+   * @param key The key to retrieve the preferences
+   */
+  getMetadata<T>(path: string, key: string): Promise<T>;
+
+  /**
+   * Save preferences to a Kaoto metadata file.
+   * @param path The path of the file being edited
+   * @param key The key to set the preferences
+   * @param metadata The preferences to be saved
+   */
+  setMetadata<T>(preferencesFilePath: string, key: string, metadata: T): Promise<void>;
```

Comment:
I think it is not the preferencesFilePath but as mentioned in the Jsdoc, the path to the file being edited